### PR TITLE
Add general-purpose interceptor hooks to cordova-plugin-approov-http …

### PR DESCRIPTION
**Pull request by johannesschneiders**
…so additional functionality can be called if required.

The hooks allow, for example, to implement certificate pinning where the trusted certificate can be changed dynamically at run-time.
As long as these hooks are not used, the plugin's functionality and behaviour are not affected.

Our aim is for these hooks to be general and useful enough that they will be accepted as contributions by you, the maintainer of cordova-plugin-approov-http.

To see what we at CriticalBlue (https://www.approov.io/) have been able to do using these hooks please see https://www.approov.io/blog/using-approov-in-your-cordova-app-the-easy-way.html
In a nutshell: we were able to integrate our mobile API protection with cordova-plugin-advanced-http without changing the way the latter is being used. Other people and organizations could use the same method to integrate their specialized functionality.

To see how we implemented the integration please refer to https://www.approov.io/blog/how-we-integrated-approov-with-cordova.html
in particular sections "Integration Approach" and "Cordova Advanced HTTP - Adding Interceptor Hooks"

### CODE REVIEW

**Comment by silkimen** [comment]
Won't merge this PR, because I'm not convinced it's a good idea to add general purpose interceptors for approov into native coding. You can use EMM / MDM providers like AirWatch, MobileIron, SOTI, etc. for implementing secure access, instead. But feel free to fork out this repository and implement your own http plugin. 👍
